pkg/healthcheck/healthcheck.go

```diff
@@ -1857,10 +1857,13 @@ func (hc *HealthChecker) checkClusterNetworks(ctx context.Context) error {
 		}
 	}
 	var badPodCIDRS []string
+	var nodesSkipped []string
+	var errStrings []string
 	var podCIDRExists bool
 	for _, node := range nodes {
 		podCIDR := node.Spec.PodCIDR
 		if podCIDR == "" {
+			nodesSkipped = append(nodesSkipped, node.Name)
```

Comment:
Is it necessary to loop through the nodes again? Can't we append to the list of errors in this main loop? I think we might be able to simplify this by not requiring an additional array just for the nodes.

```suggestion
if podCIDR == "" {
            err := fmt.Sprintf("node %s does not expose podCIDRs so clusterNetworks could not be verified", node.Name)
			errStrings = append(errStrings, err)
}
```

Reply:
Yea good call.